src/Microsoft.Diagnostics.EventFlow.Core/Implementations/DiagnosticPipelineFactory.cs

```diff
@@ -40,7 +40,13 @@ public static DiagnosticPipeline CreatePipeline(IConfiguration configuration, IH
                 healthReporter = CreateHealthReporter(configuration);
             }
             Verify.Operation(healthReporter != null, $"An instance of a health reporter could not be created and none was provider as a parameter to {nameof(CreatePipeline)} method");
-            (healthReporter as IRequireActivation)?.Activate();
+            bool? activateResult = (healthReporter as IRequireActivation)?.Activate();
```

Comment:
Let's not change the contract if not absolutely necessary.

How about throwing an exception upon activation failure? An exception can happen anyway, depending on health reporter implementation, so there is no loss of generality here.